Fix MSVC warnings, win64 bug

Hello71 opened the pull request:
- Use qSort and function pointer instead of struct operators and std::sort
  to appease MSVC.
- Simplify DFInstanceWindows::map_virtual_memory loop
- Store real base address offset instead of reading 0x400000 from PE
  header.

Hello71 committed:
Fix MSVC warnings, win64 bug

- Use qSort and function pointer instead of struct operators and std::sort
  to appease MSVC.

- Simplify DFInstanceWindows::map_virtual_memory loop

- Store real base address offset instead of reading 0x400000 from PE
  header.

splintermind commented on src/dfinstance.cpp:
```diff
@@ -1060,6 +1060,7 @@ QVector<VIRTADDR> DFInstance::get_creatures(bool report_progress){
 }
 
 bool DFInstance::is_valid_address(const VIRTADDR &addr) {
+    return true;
```
Is this intentional?

If so, this is one of two uses of the memory segments in m_regions. The other is used when scanning memory, which is virtually obsolete with df-structures and useless for a regular player.

So if we're not going to bother validating addresses, and the scanner is outdated, then map_virtual_memory won't be used anywhere and all the related scanner / virtual mem reading code can be removed entirely.

Hello71 replied:
no

Hello71 replied:
but i don't like the scanners either, and it would be great for code size if we could rip them out

splintermind replied:
I think it would be fine to remove them. It's intended as a dev-tool more than anything else, and the df-structure tools are much more useful.

Hello71 committed:
Rip out scanners, has_flag.

Hello71 committed:
Fix crash, remove some useless wrappers for "new".

Hello71 committed:
Fix Linux write_str, misc. backend improvements

- Since we don't have memory mapping anymore, hard-code the standard ELF
  base address with the assumption that DF will not get -fPIE any time
  soon.

- Rip out UCS-2 from CP437Codec and pre-register it.

- Remove some obsolete code and unnecessary includes.

Hello71 committed:
Miscellaneous minor changes.

Hello71 committed:
Refactor *nix code into DFInstanceNix, misc changes
